core/app/models/spree/credit_card.rb

```diff
@@ -163,7 +163,7 @@ def has_payment_profile?
     #   the object to ActiveMerchant.
     # @return [String] the first name on this credit card
     def first_name
-      name.to_s.split(/[[:space:]]/, 2)[0]
+      Spree::Address::Name.new(name).first_name
```

Comment:
We should deprecate this method and move the logic into `solidus_gateway`, respectively into the payment method extensions. Having ActiveMerchant related logic in core seems odd.

Reply:
I think this could be part of another PR since there are also other (unrelated) methods that we should deprecate under the same motivation. I'd say to keep this as it's minimum since it's already an important/critical change.

Reply:
Yes, I was thinking to split the work into 4-5 PRs to ease the review work, this is a proposed roadmap https://github.com/solidusio/solidus/issues/3234#issuecomment-566138962